sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -828,9 +828,11 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                                                           CancellationToken cancellationToken) => Task.Run(async () =>
         {
             var emptyPartitionContext = new EmptyPartitionContext(partitionId);
+            var connection = ConnectionFactory();
+            var consumer = CreateConsumer(ConsumerGroup, connection, ProcessingConsumerOptions);
 
-            await using var connection = ConnectionFactory();
-            await using var consumer = CreateConsumer(ConsumerGroup, connection, ProcessingConsumerOptions);
+            await using var connectionAwaiter = connection.ConfigureAwait(false);
```

Comment:
well.... that's a new one on me.  Thanks for the catch.

 You'd have thought they would have supported a better pattern for this, but it seems [not](https://github.com/dotnet/csharplang/issues/2661)